js/index.js

```diff
@@ -44,26 +44,117 @@ function renderPepperoni() {
+  sauce.setAttribute('class','sauce sauce-white');
+} else {
+  sauce.setAttribute('class','sauce');
+}
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+const crust = document.querySelector('.crust');
+if (state.glutenFreeCrust) {
+  crust.setAttribute('class','crust crust-gluten-free');
+} else {
+  crust.setAttribute('class', 'crust')
+}
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  if (state.pepperoni){
+    document.querySelector('.btn.btn-pepperoni').setAttribute('class', 'btn btn-pepperoni active');
+    
+  }
+  else {
+    document.querySelector('.btn.btn-pepperoni').setAttribute('class', 'btn btn-pepperoni');
+  }
+  if (state.mushrooms){
+    document.querySelector('.btn.btn-mushrooms').setAttribute('class', 'btn btn-mushrooms active');
+  }
+  else {
+    document.querySelector('.btn.btn-mushrooms').setAttribute('class', 'btn btn-mushrooms');
+  }
+  if (state.greenPeppers){
+    document.querySelector('.btn.btn-green-peppers').setAttribute('class', 'btn btn-green-peppers active');
+  }
+  else {
+    document.querySelector('.btn.btn-green-peppers').setAttribute('class', 'btn btn-green-peppers');
+  }
+  if (state.whiteSauce){
+    document.querySelector('.btn.btn-sauce').setAttribute('class', 'btn btn-sauce active');
+  }
+  else {
+    document.querySelector('.btn.btn-sauce').setAttribute('class', 'btn btn-sauce');
+  }
+  if (state.glutenFreeCrust){
+    document.querySelector('.btn.btn-crust').setAttribute('class', 'btn btn-crust active');
+  }
+  else {
+    document.querySelector('.btn.btn-crust').setAttribute('class', 'btn btn-crust');
+  }
 }
 
+
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+  let price = document.querySelector('aside.panel.price ul');
+  let total = 10;
+  let totalHTML = document.querySelector('aside.panel.price strong');
+  console.log(totalHTML);
+  const priceMain = 1;
+  const priceWhite = 3;
+  const priceFree = 5;
+  price.innerHTML= ``;
+  if (state.pepperoni){
+    price.innerHTML += `<li>$${priceMain} pepperoni</li>`;
+    total += priceMain;
+    totalHTML.innerHTML =`<strong>$${total}</strong>`;
+  }
+  if (state.mushrooms){
+    price.innerHTML += `<li>$${priceMain} mushrooms</li>`;
+    total += priceMain;
+    totalHTML.innerHTML =`<strong>$${total}</strong>`;
+  }
+  if (state.greenPeppers){
+    price.innerHTML += `<li>$${priceMain} green peppers</li>`;
+    total += priceMain;
+    totalHTML.innerHTML =`<strong>$${total}</strong>`;
+  }
+  if (state.whiteSauce){
+    price.innerHTML += `<li>$${priceWhite} white sauce</li>`;
+    total += priceWhite;
+    totalHTML.innerHTML =`<strong>$${total}</strong>`;
+  } 
+  if (state.glutenFreeCrust){
+  price.innerHTML += `<li>$${priceFree} gluten-free crust</li>`;
+  total += priceFree;
+  totalHTML.innerHTML =`<strong>$${total}</strong>`;
+  }
```

Comment:
Muy bien! Para evitar repetir código, puedes usar un bucle for... in de la siguiente forma:
```
for(let oneIngredient in ingredients) {
    if(state[oneIngredient]) {
       // ...
     } 
}
```